### PR TITLE
[ROCm] stateful random ops

### PR DESCRIPTION
**Pull request by ekuznetsov139**
This test enables and fixes the //tensorflow/python:stateful_random_ops test on the ROCm platform.

### CODE REVIEW

File: tensorflow/core/kernels/stateful_random_ops_cpu_gpu.h [hunk 1]
```diff
@@ -86,12 +86,19 @@ using CPUDevice = Eigen::ThreadPoolDevice;
 
 using GPUDevice = Eigen::GpuDevice;
 
+struct UpdateVariableAndFill_Philox_Arg {
+  int64 output_size;
+  int64 alg_tag_skip;
+  ScopedUnlockUnrefVar* not_used;
```
Comment (chsigg) [inline tensorflow/core/kernels/stateful_random_ops_cpu_gpu.h]: Why is this not called state_var_guard? It is used after all (`state_var_guard->Release();`).

File: tensorflow/core/kernels/stateful_random_ops_gpu.cu.cc [hunk 2]
```diff
@@ -106,6 +112,7 @@ void RngSkip_Philox<GPUDevice>::operator()(const GPUDevice& d, int64 delta,
 
 // clang-format off
 // NVCC cannot handle ">>" properly
+
```
Comment (chsigg) [inline tensorflow/core/kernels/stateful_random_ops_gpu.cu.cc]: Please revert

File: tensorflow/core/kernels/stateful_random_ops_cpu_gpu.h (hunk 1 above)
Comment (ekuznetsov139) [reply tensorflow/core/kernels/stateful_random_ops_cpu_gpu.h]: It is not used by the GPU kernel.

File: tensorflow/core/kernels/stateful_random_ops_gpu.cu.cc (hunk 2 above)
Comment (ekuznetsov139) [reply tensorflow/core/kernels/stateful_random_ops_gpu.cu.cc]: Reverted

**Review by chsigg — DISMISSED** [review]

**Comment by ravikyram** [comment]
@ekuznetsov139 Can you please check build failures? Thanks!

**Review by chsigg — APPROVED** [review]

**Pull request merged**